sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -433,9 +433,11 @@ def _configure_logger(logger):
   """ _configure_logger configures the logger such that the info level logs
   go to the stdout and the error(or above) level logs go to the stderr.
   It is important for the Jupyter notebook log rendering """
-  if logger.hasHandlers():
-    # If the logger has handlers, it has been configured, thus return immediately.
+  if hasattr(_configure_logger, 'configured'):
+    # Skip the logger configuration the second time this function
+    # is called to avoid multiple streamhandlers bound to the logger.
     return
+  setattr(_configure_logger, 'configured', 'true')
```

Comment:
You can probably just write.
```suggestion
  _configure_logger.configured = 'true'
```

Reply:
I tried to keep the code consistent using hasattr and setattr.